src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryDate.cs

```diff
@@ -719,6 +717,54 @@ public static FormulaValue Weekday(EvalVisitor runner, EvalVisitorContext contex
             return NumberOrDecimalValue(irContext, weekday);
         }
 
+        public static FormulaValue WeekNum(EvalVisitor runner, EvalVisitorContext context, IRContext irContext, FormulaValue[] args)
+        {
+            var timeZoneInfo = runner.TimeZoneInfo;
+            var startOfWeek = 0d;
+
+            if (args[0] is BlankValue)
+            {
+                return NumberOrDecimalValue(irContext, 0);
```

Comment:
Blank as the first argument should be interpreted as the "date zero" (1899-12-30), so the result may not be 0